src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py

```diff
@@ -479,60 +474,25 @@ def createEvent(digiprovMD, event_record):
     eventOutcomeDetail = etree.SubElement(eventOutcomeInformation, ns.premisBNS + "eventOutcomeDetail")
     etree.SubElement(eventOutcomeDetail, ns.premisBNS + "eventOutcomeDetailNote").text = escape(event_record.event_outcome_detail)
 
-    if event_record.linking_agent:
-        linkingAgentIdentifier = etree.SubElement(event, ns.premisBNS + "linkingAgentIdentifier")
-        etree.SubElement(linkingAgentIdentifier, ns.premisBNS + "linkingAgentIdentifierType").text = "Archivematica user pk"
-        etree.SubElement(linkingAgentIdentifier, ns.premisBNS + "linkingAgentIdentifierValue").text = str(event_record.linking_agent)
-
     # linkingAgentIdentifier
-    for agent in Agent.objects.all():
+    for agent in event_record.agents.all():
         linkingAgentIdentifier = etree.SubElement(event, ns.premisBNS + "linkingAgentIdentifier")
         etree.SubElement(linkingAgentIdentifier, ns.premisBNS + "linkingAgentIdentifierType").text = agent.identifiertype
         etree.SubElement(linkingAgentIdentifier, ns.premisBNS + "linkingAgentIdentifierValue").text = agent.identifiervalue
 
+def createAgent(digiprovMD, agent_record):
```

Comment:
Maybe keep these as separate parameters so we can call `values` above and avoid the need to instantiate an `Agent` object? We're not using the stuff that a model object would give us.